lib.php

```diff
@@ -46,7 +46,7 @@ function tool_mfa_after_require_login() {
 
         $clearurl = str_replace($CFG->wwwroot, '', $ME);
 
-        if (strpos($clearurl, '/admin/tool/mfa/') !== 0) {
+        if (strpos($clearurl, '/admin/tool/mfa/auth.php') !== 0) {
```

Comment:
Can you please move the wantsurl logic down into here, we should only set it if we do end up actually redirecting

Can you also please refactor this into a 'should_redirect($url)' method and wrap some unit tests around it.

Right now the logic is simple but its going to get more complex. As a starting point hack you account so that user_not_fully_set_up() returns false which will try to redirect you to edit your profile before this callback fires, which will then start an infinite redirect loop here

Reply:
Other tests:
* if you are AJAX never redirect
* if you are a webservice all never redirect